miio/airfresh_t2017.py

```diff
@@ -134,32 +162,60 @@ def control_speed(self) -> int:
     @property
     def dust_filter_life_remaining(self) -> int:
         """Remaining dust filter life in percent."""
-        return self.data["filter_intermediate"]
+        if (
+            "filter_intermediate" in self.data
+            and self.data["filter_intermediate"] is not None
+        ):
+            return self.data["filter_intermediate"]
+
+        # Filter property of the A1
+        if "filter_rate" in self.data and self.data["filter_rate"] is not None:
+            return self.data["filter_rate"]
 
     @property
     def dust_filter_life_remaining_days(self) -> int:
         """Remaining dust filter life in days."""
-        return self.data["filter_inter_day"]
+        if (
+            "filter_inter_day" in self.data
+            and self.data["filter_inter_day"] is not None
+        ):
+            return self.data["filter_inter_day"]
+
+        # Filter property of the A1
+        if "filter_day" in self.data and self.data["filter_day"] is not None:
+            return self.data["filter_day"]
 
     @property
-    def upper_filter_life_remaining(self) -> int:
+    def upper_filter_life_remaining(self) -> Optional[int]:
         """Remaining upper filter life in percent."""
-        return self.data["filter_efficient"]
+        if (
+            "filter_efficient" in self.data
+            and self.data["filter_efficient"] is not None
+        ):
+            return self.data["filter_efficient"]
+
+        return None
 
     @property
-    def upper_filter_life_remaining_days(self) -> int:
+    def upper_filter_life_remaining_days(self) -> Optional[int]:
         """Remaining upper filter life in days."""
-        return self.data["filter_effi_day"]
+        if "filter_effi_day" in self.data and self.data["filter_effi_day"] is not None:
+            return self.data["filter_effi_day"]
+
+        return None
 
     @property
     def ptc(self) -> bool:
         """Return True if PTC is on."""
         return self.data["ptc_on"]
 
     @property
-    def ptc_level(self) -> int:
+    def ptc_level(self) -> Optional[int]:
         """PTC level."""
-        return PtcLevel(self.data["ptc_level"])
```

Comment:
Maybe this sort of structure would work for all properties that return an `Optional[SomeEnumOrContainerType]` (btw, the current type hint is wrong):
```
try:
    return PtcLevel(self.data["ptc_level"])
except KeyError:
    return None